crates/env_filter/src/parser.rs

```diff
@@ -22,8 +24,36 @@ impl ParseResult {
     fn add_error(&mut self, message: String) {
         self.errors.push(message);
     }
+
+    pub(crate) fn ok(self) -> Result<(Vec<Directive>, Option<FilterOp>), ParseError> {
+        if self.errors.is_empty() {
+            Ok((self.directives, self.filter))
+        } else {
+            Err(ParseError {
+                details: self.errors,
```

Comment:
Unsure how I feel about printing the details comma separated.  Should we just return the first?

Reply:
I would argue that the more details an error message provides, the better. In this case, we already allocate for `String` message in the error, so giving more details is doesn't have downsides in my opinion. It's often frustrating when a system returns only first error of many when it easily can save you several iterations of fixing your code by simply telling you all your errors right away.

If you find `Vec` in the error struct inappropriate, we could join strings here instead of doing it in the `Display` (although I think it's better to keep messages separated in the error struct).

Reply:
I'm not worried about performance but the usability of concatenating the errors. imo we should only return one.  If someone finds a concrete use case where this is inhibiting usability enough, let's evaluate a "error recovery" API that returns a `Vec<ParseError>` (or some other related approach).

Reply:
All right, I've made the error contain the first error message.